diff -uNr libzypp-bindings-0.5.50/swig/Arch.i libzypp-bindings-0.5.50.new/swig/Arch.i
--- libzypp-bindings-0.5.50/swig/Arch.i	2011-10-19 00:35:54.000000000 +0800
+++ libzypp-bindings-0.5.50.new/swig/Arch.i	2012-01-29 16:11:03.839382406 +0800
@@ -84,6 +84,22 @@
    * builtin: s390s (zSeries 64 bit)
    */
   static Arch s390x() { return zypp::Arch_s390x; }
+   /*
+   * builtin: armv7tnhl
+   */
+  static Arch armv7tnhl() { return zypp::Arch_armv7tnhl; }
+  /*
+   * builtin: armv7thl
+   */
+  static Arch armv7thl() { return zypp::Arch_armv7thl; }
+  /*
+   * builtin: armv7nhl
+   */
+  static Arch armv7nhl() { return zypp::Arch_armv7nhl; }
+  /*
+   * builtin: armv7hl
+   */
+  static Arch armv7hl() { return zypp::Arch_armv7hl; }
   /*
    * builtin: armv7tnhl
    */
